File: src/main/java/com/burgstaller/okhttp/CachingUtils.java
package com.burgstaller.okhttp;

import okhttp3.HttpUrl;

public final class CachingUtils {

    /**
     * Get key to be used for storing cached auth responses, e.g.
     * http:myhost.com:8080
     */
    public static String getCachingKey(HttpUrl url) {
        if (url == null)
            return null;
        return url.scheme() + ":" + url.host() + ":" + url.port();
    }

}

File: src/main/kotlin/com/burgstaller/okhttp/CachingUtils.kt
package com.burgstaller.okhttp

import okhttp3.HttpUrl

object CachingUtils {
  /**
   * Get key to be used for storing cached auth responses, e.g.
   * http:myhost.com:8080
   */
  fun defaultCacheKey(url: HttpUrl): String {
    return "${url.scheme()}:${url.host()}:${url.port()}"
  }
}
